src/client/__tests__/e2e/pages/blotter.component.ts

```diff
@@ -0,0 +1,45 @@
+import { by, ElementFinder, ProtractorBrowser } from 'protractor'
+import { mapAsync } from '../utils/async.utils'
+import { waitForElementToBePresent } from '../utils/browser.utils'
+
+export interface BlotterHeader {
+  id: string
+  label: string
+}
+
+export class BlotterComponent {
+  tradesTable: Record<string, Record<string, ElementFinder>>
+
+  constructor(private browser: ProtractorBrowser, public root: ElementFinder) {
+    this.tradesTable = {
+      executedTrades: {
+        tradeID: root.element(by.xpath('//*[@id="root"]/div/div/div[2]/div/div[1]/div[2]/div/div[2]/div/div/div[2]/div/div/div[1]/div/div[3]/div[2]/div/div/div[1]/div[2]')),
```

Comment:
These locators are quite flakey as they heavily rely on the DOM structure.

It might be worth adding a `data-testid="xxx"` attribute to the elements you want to target in tests and then using `by.css("[id="root"] [data-testid=xxx]")` in the locators

(Also it's worth noting that `by.xpath` is not efficient, `by.css` is much faster because it can use the browser's `querySelectorAll`)

Reply:
We can't implement data-qa id's for blotter components but i am gonna come with somthing better than xpath

Reply:
@BenGladman i removed the validations using these locators for the blotter section as there are unstable with css selectors until we find the better solutions for them.

Reply:
Could you use something like:
`by.css('[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="tradeId"]`

that looks reasonably resilient to me

Reply:
i will try i am not really sure

Reply:
it worked thanks i will update them all